pilot/pkg/xds/proxy_dependencies.go

```diff
@@ -19,12 +19,15 @@ import (
 	"istio.io/istio/pilot/pkg/model"
 	"istio.io/istio/pkg/config/host"
 	"istio.io/istio/pkg/config/schema/kind"
+	"istio.io/istio/pkg/util/sets"
 )
 
-// configKindAffectedProxyTypes contains known config types which may affect certain node types.
-var configKindAffectedProxyTypes = map[kind.Kind][]model.NodeType{
-	kind.Gateway: {model.Router},
-	kind.Sidecar: {model.SidecarProxy},
+// UnAffectedConfigKinds contains config types which does not affect certain proxy types.
+var UnAffectedConfigKinds = map[model.NodeType]sets.Set[kind.Kind]{
```

Comment:
Seems not equal to `configKindAffectedProxyTypes`,  what about ztunel/waypoint
@howardjohn Can you confirm

Reply:
https://github.com/istio/istio/blob/21f450960ce820ada1e125529a843ac8fb55f054/pilot/pkg/xds/proxy_dependencies.go#L37 - it is not used

Reply:
may be it is little inverted - but certainly simplifies code. Otherwise I am having hard time understanding it